### PR TITLE
wip: scaling support mostly working

### PR DESCRIPTION
**Pull request by wash2**

### CODE REVIEW

File: src/shared_state.rs [hunk 1]
```diff
@@ -51,8 +51,34 @@ impl<W: WrapperSpace + 'static> GlobalState<W> {
...
+    /// this should be called when the scale factor of a surface changes
+    pub fn scale_factor_changed(&mut self, surface: &WlSurface, scale_factor: f64, legacy: bool) {
+        if legacy && self.client_state.fractional_scaling_manager.is_some() {
+            return;
+        }
+        for tracked_surface in &self.client_state.proxied_layer_surfaces {
+            if tracked_surface.3.wl_surface() == surface {
+                if legacy {
+                    dbg!("setting buffer scale i guess idk");
+                    surface.set_buffer_scale(scale_factor as i32);
```
Comment (Drakulix) [inline src/shared_state.rs:69]: We probably want to round up in this case.
```suggestion
                    surface.set_buffer_scale(scale_factor.ceil() as i32);
```

Comment (wash2) [reply src/shared_state.rs:69]: Ah thanks, I assume you mean ceil?

Comment (wash2) [reply src/shared_state.rs:69]: I think in this case, the scale_factor is already an integer that was converted to a double.

Comment (Drakulix) [reply src/shared_state.rs:69]: Yes, that is what I meant. In that case this is probably the wrong place to do this. :sweat_smile:

**Pull request merged**